php/libraries/Utility.class.inc

```diff
@@ -558,7 +558,7 @@ class Utility
      */
     static function asArray($var)
     {
-        if (!is_array($var)) {
+        if (!is_array($var) && !empty($var)) {
```

Comment:
Why is this changed? This seems likely to have unexpected impacts in different parts of the code. Can't you add the empty check in the place that you're using it to be safe?